ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs

```diff
@@ -298,3 +309,97 @@ unqalDP paren =
+    Maybe (Located [LIE GhcPs]) ->
+    DynFlags ->
+    TransformT (Either String) (LImportDecl GhcPs)
+extendHiding symbol (L l idecls) mlies df = do
+    L l' lies <- case mlies of
+        Nothing -> flip L [] <$> uniqueSrcSpanT
+        Just pr -> pure pr
+    let hasSibling = not $ null lies
+    src <- uniqueSrcSpanT
+    top <- uniqueSrcSpanT
+    rdr <- liftParseAST df symbol
+    let lie = L src $ IEName rdr
+        x = L top $ IEVar noExtField lie
+        singleHide = L l' [x]
+    when (isNothing mlies) $ do
+        addSimpleAnnT
+            singleHide
+            dp00
+            [ (G AnnHiding, DP (0, 1))
+            , (G AnnOpenP, DP (0, 1))
+            , (G AnnCloseP, DP (0, 0))
+            ]
+    addSimpleAnnT x (DP (0, 0)) []
+    addSimpleAnnT rdr dp00 $ unqalDP $ isOperator $ unLoc rdr
+    if hasSibling
+        then when hasSibling $ do
+            addTrailingCommaT x
+            addSimpleAnnT (head lies) (DP (0, 1)) []
+            unless (null $ tail lies) $
+                addTrailingCommaT (head lies) -- Why we need this?
+        else forM_ mlies $ \lies0 -> do
+            transferAnn lies0 singleHide id
+    return $ L l idecls {ideclHiding = Just (True, L l' $ x : lies)}
+    where
+        isOperator = not . all isAlphaNum . occNameString . rdrNameOcc
+
+deleteFromImport ::
+    String ->
+    LImportDecl GhcPs ->
+    Located [LIE GhcPs] ->
+    DynFlags ->
+    TransformT (Either String) (LImportDecl GhcPs)
+deleteFromImport symbol (L l idecl) llies@(L lieLoc lies) _ =do
+    let edited = L lieLoc deletedLies
+        lidecl' = L l $ idecl
+            { ideclHiding = Just (False, edited)
+            }
+    when (not (null lies) && null deletedLies) $ do
+        transferAnn llies edited id
+        addSimpleAnnT edited dp00
+            [(G AnnOpenP, DP (0, 1))
+            ,(G AnnCloseP, DP (0,0))
+            ]
+    pure lidecl'
+    where
+        deletedLies =
+            mapMaybe killLie lies
+        killLie :: LIE GhcPs -> Maybe (LIE GhcPs)
+        killLie v@(L _ (IEVar _ (L _ (rawIEWrapName -> nam))))
+            | nam == symbol  = Nothing
+            | otherwise = Just v
+        killLie v@(L _ (IEThingAbs _ (L _ (rawIEWrapName -> nam))))
+            | nam == symbol = Nothing
+            | otherwise = Just v
+
+        killLie (L lieL (IEThingWith xt ty@(L _ (rawIEWrapName -> nam)) wild cons flds))
+            | nam == symbol = Nothing
+            | otherwise = Just $
+                L lieL $ IEThingWith xt ty wild
+                  (filter ((/= symbol) . rawIEWrapName . unLoc) cons)
+                  (filter ((/= symbol) . unpackFS . flLabel . unLoc) flds)
+        killLie v = Just v
+
+-- This must not belong here?
+rawIEWrapName :: IEWrappedName RdrName -> String
+rawIEWrapName (IEName (L _ nam)) = occNameString $ rdrNameOcc nam
+rawIEWrapName (IEPattern (L _ nam)) = occNameString $ rdrNameOcc nam
+rawIEWrapName (IEType (L _ nam)) = occNameString $ rdrNameOcc nam
```

Comment:
isn't this just `prettyPrint`?

Reply:
Yes, it must be equivalent to `prettyPrint` on `OccName`. One difference is that if we apply `prettyPrint` directly to `IEWrappedName`, then we have some modifiers added around identifiers. e.g.

```hs
>>> import Development.IDE.GHC.Util
>>> prettyPrint (Unqual $ mkVarOcc "++")
"++"

>>> prettyPrint $ IEType $ L (UnhelpfulSpan "<input>") $ Unqual $ mkVarOcc "++"
"type (++)"
```

Another difference is that we need unqualified names here. If we apply `prettyPrint` to `RdrName`, we get, e.g.

```
-- >>> prettyPrint $ Qual (mkModuleName "Bar") $ mkVarOcc "++"
-- "Bar.++"
```

Hence we need to strip IEWrapping and qualifiers here.

Reply:
you can use `showNameWithoutUniques` which will remove the qualifiers, or you can create your own render style

Reply:
It seems that `showNameWithoutUniques` doesn't remove qualifier, only Uniques generated by GHC. Anyway, it gives the expected result on OccName, and returning type is Text, so I decided to use `showNameWithoutUniques` inside `rawIEWrapName` and move to `Development.IDE.Spans.Common.unqualIEWrapName`.